src/plugins/lua/mid.lua

```diff
@@ -0,0 +1,112 @@
+--[[
+Copyright (c) 2016, Alexander Moisseev <moiseev@mezonplus.ru>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+]]--
+
+--[[
+MID plugin - suppress INVALID_MSGID and MISSING_MID for messages originating
+from listed valid DKIM domains with missed or known proprietary Message-IDs
+]]--
+
+local rspamd_logger = require "rspamd_logger"
+local rspamd_regexp = require "rspamd_regexp"
+
+local settings = {
+  url = '',
+  symbol_known_mid = 'KNOWN_MID',
+  symbol_known_no_mid = 'KNOWN_NO_MID',
+  symbol_invalid_msgid = 'INVALID_MSGID',
+  symbol_missing_mid = 'MISSING_MID',
+  symbol_dkim_allow = 'R_DKIM_ALLOW',
+  csymbol_invalid_msgid_allowed = 'INVALID_MSGID_ALLOWED',
+  csymbol_missing_mid_allowed = 'MISSING_MID_ALLOWED',
+}
+
+local maps = {}
+
+local function known_mid_cb(task)
+  local re = {}
+  local header = task:get_header('Message-Id')
+  local das = task:get_symbol(settings['symbol_dkim_allow'])
+  if das and das[1] and das[1]['options'] then
+    for _,dkim_domain in ipairs(das[1]['options']) do
+      for _,map in ipairs(maps) do
+        local v = map:get_key(dkim_domain)
+        if v then
+          if v == '' then
+            if not header then
+              task:insert_result(settings['symbol_known_no_mid'], 1, dkim_domain)
+              return
+            end
+          else
+            re[dkim_domain] = rspamd_regexp.create_cached(v)
```

Comment:
For the first issue, let's assume that incorrect regexp is a warning and add symbol in this case (like if it was no regexp).
For the second issue, I'll think about expiration to the regexp cache (so you can leave `create_cached` safely).